src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/ConstructorWriter.java

```diff
@@ -189,13 +189,15 @@ protected Content getConstructorHeaderContent(ExecutableElement constructor) {
         Content content = new ContentBuilder();
         var heading = HtmlTree.HEADING(Headings.TypeDeclaration.MEMBER_HEADING,
                 Text.of(name(constructor)));
+        HtmlId memberAnchor = htmlIds.forMember(constructor);
         HtmlId erasureAnchor = htmlIds.forErasure(constructor);
-        if (erasureAnchor != null) {
+        if (erasureAnchor != null
+                && !erasureAnchor.name().equals(memberAnchor.name())) {
```

Comment:
Is this redundant now that the erasure handling is encapsulated within forMember?

Reply:
No, it's not redundant. `forMember` returns either the simple id or the erased id. `forErasure` always returns the erased id. Therefore, if `forMember` returns the erased id, there will be a duplicating id, which is a no-no.